src/xds.rs

```diff
@@ -107,15 +107,46 @@ mod google {
     }
 }
 
-const ENDPOINT_TYPE: &str = "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
-const CLUSTER_TYPE: &str = "type.googleapis.com/envoy.config.cluster.v3.Cluster";
-const LISTENER_TYPE: &str = "type.googleapis.com/envoy.config.listener.v3.Listener";
-
-pub use xds::*;
+pub const CLUSTER_TYPE: &str = "type.googleapis.com/envoy.config.cluster.v3.Cluster";
+pub const ENDPOINT_TYPE: &str =
+    "type.googleapis.com/envoy.config.endpoint.v3.ClusterLoadAssignment";
+pub const EXTENSION_CONFIG_TYPE: &str =
+    "type.googleapis.com/envoy.config.core.v3.TypedExtensionConfig";
+pub const LISTENER_TYPE: &str = "type.googleapis.com/envoy.config.listener.v3.Listener";
+pub const ROUTE_TYPE: &str = "type.googleapis.com/envoy.config.route.v3.RouteConfiguration";
+pub const RUNTIME_TYPE: &str = "type.googleapis.com/envoy.service.runtime.v3.Runtime";
+pub const SCOPED_ROUTE_TYPE: &str =
+    "type.googleapis.com/envoy.config.route.v3.ScopedRouteConfiguration";
+pub const SECRET_TYPE: &str =
+    "type.googleapis.com/envoy.extensions.transport_sockets.tls.v3.Secret";
+pub const VIRTUAL_HOST_TYPE: &str = "type.googleapis.com/envoy.config.route.v3.VirtualHost";
```

Comment:
It depends on the case, I think with this one adding `skip` wouldn't add much here, because it would still be having two lines for each one. If we want to make this clearer, I could add this macro here instead.

```rust
macro_rules! type_urls {
    ($($base_url:literal : {$($const_name:ident = $type_url:literal),+ $(,)?})+) => {
        $(
            $(
                pub const $const_name : &str = concat!($base_url, "/", $type_url);
            )+
        )+
    }
}

type_urls! {
    "type.googleapis.com": {
        CLUSTER_TYPE = "envoy.config.cluster.v3.Cluster",
        ENDPOINT_TYPE = "envoy.config.endpoint.v3.ClusterLoadAssignment",
        EXTENSION_CONFIG_TYPE = "envoy.config.core.v3.TypedExtensionConfig",
        LISTENER_TYPE = "envoy.config.listener.v3.Listener",
        ROUTE_TYPE = "envoy.config.route.v3.RouteConfiguration",
        RUNTIME_TYPE = "envoy.service.runtime.v3.Runtime",
        SCOPED_ROUTE_TYPE = "envoy.config.route.v3.ScopedRouteConfiguration",
        SECRET_TYPE = "envoy.extensions.transport_sockets.tls.v3.Secret",
        VIRTUAL_HOST_TYPE = "envoy.config.route.v3.VirtualHost",
    }
}
```

Reply:
>  Thought it would make rustfmt not split lines if they're joined to a single one per url.

It would, but there would now also have to be a `#[rustfmt::skip]` on top of each one, which IMO would it harder to read than just having the lines split.

Also FWIW there's no real overhead for a macro like that, it's just very simple expansion.

Reply:
Oh yeah my bad, didn't test this properly, thought it would affect the whole group.

I mostly meant syntactically and to know what's going on without having to look what the macro does. I think it's just as good to have a regular `pub const` with the `&str` value and live with the line breaks but I don't have a very strong opinion if you want to go the macro way. Do what you think is nicest 😃